caikit_nlp/toolkit/text_generation/model_run_utils.py

```diff
@@ -233,7 +238,9 @@ def generate_text_func(
         for g in generate_ids
     ]
 
-    if generate_ids[0][-1].item() == eos_token:
+    if (eos_token and tokenizer.decode(generate_ids[0, -1].item()) == eos_token) or (
```

Comment:
Also adding an extra check for the `eos_token`, since it's actually an `Optional[str]`, falling back to `tokenizer.eos_token_id` if it is not defined. Hopefully this is the correct behaviour.